content/article/enhancing-midi-hardware-with-perl.md

```diff
@@ -0,0 +1,507 @@
+{
+    "title"       : "Enhancing your MIDI devices with Perl",
+    "authors"     : ["john-barrett"],
+    "date"        : "2025-01-30",
+    "tags"        : ["midi","music"],
+    "draft"       : true,
+    "thumbnail"   : "/images/enhancing-midi-hardware-with-perl/midicamel.png",
+    "description" : "Add functions and features to MIDI devices by routing them via Perl",
+    "categories"  : "data"
+}
+
+## Introduction
+
+These days, even modestly priced MIDI hardware comes stuffed with features.
+You should expect a budget device to provide at least some of clock, sequencer,
+arpeggiator, chord voicing, Digital Audio Workstation (DAW) integration, and
+transport control features.
+
+Fitting all
+this into a small device's form factor may result in some amount of compromise
+— perhaps modes aren't easily combined, or some amount of menu diving is
+required to switch between modes. Your device may even lack the precise
+functionality you require.
+
+This post will walk through the implementation of a pair of features to augment
+those found in a MIDI keyboard — a M-Audio Oxygen Pro 61 in this case, though
+the principle should apply to any device.
+
+## Feature 1 : Pedal Tone
+
+A recent video by [Polarity Music opened with some exploration of using a pedal tone](https://youtu.be/Cu1FKGp6pKQ)
+in Bitwig to compose progressions. A pedal tone (or pedal note, or pedal point)
+is a sustained single note, over which other potentially dissonant parts are
+played. In this case, the pedal tone was gated by the keyboard, and the fifth
+interval of the played note was added resulting in a three note chord for a
+single played note. This simple setup resulted in some dramatic progressions.
+
+There are, of course, ways to achieve this effect in other DAW software. I was
+able to use Patcher within FL Studio to achieve a similar result with two
```

Comment:
What is Patcher? Link? :)
Link to "What is FL Studio" too?
And what is "VFX Key Mapper"?